src/main/kotlin/io/icure/md/client/apis/impl/HealthcareElementApiImpl.kt

```diff
@@ -95,15 +106,80 @@ class HealthcareElementApiImpl(private val api: MedTechApi) : HealthcareElementA
     }
 
     override suspend fun getHealthcareElement(healthcareElementId: String): HealthcareElement {
-        val currentUser = api.userApi().getCurrentUser()
-        val localCrypto = api.localCrypto()
+        val currentUser = medTechApi.baseUserApi.getCurrentUser()
+        val localCrypto = medTechApi.localCrypto
 
-        return api.healthElementApi()
+        return medTechApi.baseHealthElementApi
             .getHealthElement(currentUser, healthcareElementId, healthElementCryptoConfig(localCrypto))
             .toHealthcareElement()
     }
 
     override suspend fun matchHealthcareElement(filter: Filter<HealthcareElement>): List<String> {
-        return api.healthElementApi().matchHealthElementsBy(filter.toAbstractFilterDto())
+        return medTechApi.baseHealthElementApi.matchHealthElementsBy(filter.toAbstractFilterDto())
+    }
+
+    override suspend fun giveAccessTo(healthcareElement: HealthcareElement, delegateTo: String): HealthcareElement {
+        val localCrypto = medTechApi.localCrypto
+        val currentUser = medTechApi.baseUserApi.getCurrentUser()
+        val dataOwnerId = currentUser.dataOwnerId()
+
+        if (!healthcareElement.systemMetaData!!.delegations.keys.any { it == dataOwnerId }) {
+            throw IllegalStateException("DataOwner $dataOwnerId does not have the right to access it ${healthcareElement.id}")
+        }
+
+        if (healthcareElement.systemMetaData.delegations.keys.any { it == delegateTo }) {
+            return healthcareElement
+        }
+
+        val ccHealthElement = healthElementCryptoConfig(localCrypto)
+
+        return healthcareElement.toHealthcareElementDto().let { healthElementDto ->
```

Comment:
For me you can directly do this work on MedTech HealthcareElement (not on icure-reactive-kotlin-client dto) and call the createOrModifyHealthElement, so that all the exceptions would be covered as well (like in PatientLogicImpl)